[BUG]: Compilation failure since 1.7.3669

### Describe the Bug

Pcsx2 has stopped building since 1.7.3669 with multiple undefined references, see log: https://paste.xinu.at/k26mTd0wZCwFKjyz/
Disabling LTO didn't help.

### Reproduction Steps

* using the cmake command from the following build recipe: https://github.com/archlinux/svntogit-community/blob/packages/pcsx2/trunk/PKGBUILD
* observe link failures from the above logs

### Expected Behavior

_No response_

### PCSX2 Revision

v1.7.3669

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Arch Linux

Decided to experiment with the SIMD option. I was able to get it to compile if I enabled Clang. Not GCC even with LDD enabled. Though Clang + LTO is incredibly deadly with the AUR, https://aur.archlinux.org/packages/pcsx2-git

```
[1/527] Linking C static library 3rdparty/rcheevos/librcheevos.a
FAILED: 3rdparty/rcheevos/librcheevos.a 
: && /usr/bin/cmake -E rm -f 3rdparty/rcheevos/librcheevos.a && "CMAKE_C_COMPILER_AR-NOTFOUND" cr 3rdparty/rcheevos/librcheevos.a [...]
/bin/sh: line 1: CMAKE_C_COMPILER_AR-NOTFOUND: command not found
[2/527] Linking CXX static library 3rdparty/discord-rpc/libdiscord-rpc.a
FAILED: 3rdparty/discord-rpc/libdiscord-rpc.a 
: && /usr/bin/cmake -E rm -f 3rdparty/discord-rpc/libdiscord-rpc.a && "CMAKE_CXX_COMPILER_AR-NOTFOUND" cr 3rdparty/discord-rpc/libdiscord-rpc.a [...]
/bin/sh: line 1: CMAKE_CXX_COMPILER_AR-NOTFOUND: command not found
ninja: build stopped: subcommand failed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Setting it to a range tells it that it supports the features on the newer
 # versions as well, avoiding setting policies.
-cmake_minimum_required(VERSION 3.11...3.22)
+cmake_minimum_required(VERSION 3.12...3.24)
 
 if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
 	message(FATAL_ERROR "PCSX2 does not support in-tree builds.  Please make a build directory that is not the PCSX2 source directory and generate your CMake project there using either `cmake -B build_directory` or by running cmake from the build directory.")

--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -6,7 +6,12 @@ add_library(PCSX2_FLAGS INTERFACE)
 if (NOT PCSX2_CORE)
 	add_executable(PCSX2)
 else()
-	add_library(PCSX2)
+	if(DISABLE_ADVANCE_SIMD OR LTO_PCSX2_CORE)
+		# Fixes issues with some compiler + linker combinations
+		add_library(PCSX2 OBJECT)
+	else()
+		add_library(PCSX2)
+	endif()
 	target_compile_definitions(PCSX2_FLAGS INTERFACE
 		"PCSX2_CORE"
 	)

